multiprofit/objects.py

```diff
@@ -1727,26 +1739,106 @@ def isgaussian(self):
 
     def __init__(self, fluxes, name=""):
         for i, param in enumerate(fluxes):
-            if not isinstance(param, Parameter):
+            if not isinstance(param, FluxParameter):
                 raise TypeError(
-                    "Component param[{:d}] (type={:s}) is not an instance of {:s}".format(
-                        i, str(type(param)), str(type(Parameter)))
+                    "Component flux[{:d}] (type={:s}) is not an instance of {:s}".format(
+                        i, str(type(param)), str(type(FluxParameter)))
                 )
         self.fluxes = fluxes
         self.name = name
 
 
+class Ellipse:
+    def getcovariance(self):
+        cov = self.rho*self.sigma_x*self.sigma_y
+        return np.array([[self.sigma_x**2, cov], [cov, self.sigma_y**2]])
+
+    def __init__(self, sigma_x, sigma_y, rho):
+        self.sigma_x = sigma_x
+        self.sigma_y = sigma_y
+        self.rho = rho
+
+
+class EllipseParameters(Ellipse):
+    def getcovariance(self):
+        sigma_x = self.sigma_x.getvalue(transformed=False)
+        sigma_y = self.sigma_y.getvalue(transformed=False)
+        cov = self.rho.getvalue(transformed=False)*sigma_x*sigma_y
+        return np.array([[sigma_x**2, cov], [cov, sigma_y**2]])
```

Comment:
Any way to reduce code duplication between here and the parent class?
My 2x2 vs 1x3 shape question again.